src/Microsoft.Data/DataFrame.cs

```diff
@@ -31,6 +32,10 @@ public DataFrame()
         public DataFrame(IList<BaseColumn> columns)
         {
             _table = new DataFrameTable(columns);
+            foreach (BaseColumn column in columns)
+            {
+                column.DataFrame = this; 
```

Comment:
Is this really safe? `this` is being constructed here, so I'm not sure.

Reply:
Maybe instead of having `BaseColumn` hold onto the `DataFrame` for the lifetime, maybe passing in the `DataFrame` to `GroupBy()` would work better?

That way you could have a `Column` object without a DataFrame.

---
In reply to: [305510252](https://github.com/dotnet/corefxlab/pull/2696#discussion_r305510252) [](ancestors = 305510252)